sklearn/calibration.py

```diff
@@ -311,145 +320,153 @@ def _more_tags(self):
 
-        idx_pos_class = self.label_encoder_.\
-            transform(self.base_estimator.classes_)
+def _fit_calibrator(clf_fitted, label_encoder_, method, X, y,
+                    sample_weight=None):
+    """Fit calibrator(s) and return a `_CalibratedClassiferPipeline`
+    instance.
 
-        return df, idx_pos_class
+    Output of the `decision_function` method of the `clf_fitted` is used for
+    calibration. If this method does not exist, `predict_proba` method used.
 
-    def fit(self, X, y, sample_weight=None):
-        """Calibrate the fitted model
+    Parameters
+    ----------
+    clf_fitted : Estimator instance
+        Fitted classifier.
 
-        Parameters
-        ----------
-        X : array-like, shape (n_samples, n_features)
-            Training data.
+    label_encoder_ : LabelEncoder instance
+        LabelEncoder instance fitted on all the targets.
 
-        y : array-like, shape (n_samples,)
-            Target values.
+    method : {'sigmoid', 'isotonic'}
+        The method to use for calibration.
 
-        sample_weight : array-like of shape (n_samples,), default=None
-            Sample weights. If None, then samples are equally weighted.
+    X : array-like
+        Sample data used to calibrate predictions.
 
-        Returns
-        -------
-        self : object
-            Returns an instance of self.
-        """
+    y : ndarray, shape (n_samples,)
+        The targets.
 
-        self.label_encoder_ = LabelEncoder()
-        if self.classes is None:
-            self.label_encoder_.fit(y)
-        else:
-            self.label_encoder_.fit(self.classes)
+    sample_weight : ndarray, shape (n_samples,), default=None
+        Sample weights. If `None`, then samples are equally weighted.
 
-        self.classes_ = self.label_encoder_.classes_
-        Y = label_binarize(y, classes=self.classes_)
+    Returns
+    -------
+    pipeline : _CalibratedClassiferPipeline instance
+    """
+    Y = label_binarize(y, classes=label_encoder_.classes_)
+    df, idx_pos_class = _get_predictions(clf_fitted, X, label_encoder_)
 
-        df, idx_pos_class = self._preproc(X)
-        self.calibrators_ = []
+    if method == 'isotonic':
+        calibrator = IsotonicRegression(out_of_bounds='clip')
+    elif method == 'sigmoid':
+        calibrator = _SigmoidCalibration()
 
-        for k, this_df in zip(idx_pos_class, df.T):
-            if self.method == 'isotonic':
-                calibrator = IsotonicRegression(out_of_bounds='clip')
-            elif self.method == 'sigmoid':
-                calibrator = _SigmoidCalibration()
-            else:
-                raise ValueError('method should be "sigmoid" or '
-                                 '"isotonic". Got %s.' % self.method)
-            calibrator.fit(this_df, Y[:, k], sample_weight)
-            self.calibrators_.append(calibrator)
+    calibrated_classifiers = []
+    for idx, this_df in zip(idx_pos_class, df.T):
```

Comment:
I'd suggest using `class_idx` instead of just `idx`. (`k` is commonly used to define a class so it'd be fine too though maybe more cryptic )

Reply:
Ah I didn't realise the `k` terminology otherwise I would have kept it. I will use `class_idx` for others like me who aren't familiar with the convention.